Source/Core/Core/PowerPC/MMU.cpp

```diff
@@ -182,40 +196,30 @@ static T ReadFromHardware(u32 em_address)
         GenerateDSIException(em_address, false);
       return 0;
     }
-    if ((em_address & (HW_PAGE_SIZE - 1)) > HW_PAGE_SIZE - sizeof(T))
-    {
-      // This could be unaligned down to the byte level... hopefully this is rare, so doing it this
-      // way isn't too terrible.
-      // TODO: floats on non-word-aligned boundaries should technically cause alignment exceptions.
-      // Note that "word" means 32-bit, so paired singles or doubles might still be 32-bit aligned!
-      u32 em_address_next_page = (em_address + sizeof(T) - 1) & ~(HW_PAGE_SIZE - 1);
-      auto addr_next_page = TranslateAddress<flag>(em_address_next_page);
-      if (!addr_next_page.Success())
-      {
-        if (flag == XCheckTLBFlag::Read)
-          GenerateDSIException(em_address_next_page, false);
-        return 0;
-      }
-      T var = 0;
-      u32 addr_translated = translated_addr.address;
-      for (u32 addr = em_address; addr < em_address + sizeof(T); addr++, addr_translated++)
-      {
-        if (addr == em_address_next_page)
-          addr_translated = addr_next_page.address;
-        var = (var << 8) | ReadFromHardware<flag, u8, true>(addr_translated);
-      }
-      return var;
-    }
     em_address = translated_addr.address;
   }
 
-  // TODO: Make sure these are safe for unaligned addresses.
+  if (flag == XCheckTLBFlag::Read && (em_address & 0xF8000000) == 0x08000000)
+  {
+    if (em_address < 0x0c000000)
+      return EFB_Read(em_address);
+    else
+      return (T)Memory::mmio_mapping->Read<typename std::make_unsigned<T>::type>(em_address);
```

Comment:
Missed in previous review (my bad)

```suggestion
      return (T)Memory::mmio_mapping->Read<std::make_unsigned_t<T>>(em_address);
```